### PR TITLE
fixes integrations

### PR DESCRIPTION
**Pull request by gmmeyer**
This started out as a small bug fix and balooned as a found other bugs.

This fixes a small syntax error, first. It removes a few debugging lines, such as when I was ls -l the integrations directory.

It fixes adding the config files, it changes the sdk directory to integrations, rather than 3rd-party, and it makes the pip install more fault tolerant.

Finally, it (hopefully) fixes the license file conflict.

### CODE REVIEW

File: resources/datadog-integrations/project.rb.erb [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+
 name 'dd-check-<%= name %>'
 maintainer 'Datadog Packages <package@datadoghq.com>'
 homepage 'http://www.datadoghq.com'
 install_dir '/opt/datadog-agent'
 
 build_version '<%= version %>'
 build_iteration <%= build_iteration %>
-json_manifest_path "#{install_dir}/3rd-party/<%= name %>/version-manifest.json"
+json_manifest_path "/opt/datadog-agent/integrations/<%= name %>/version-manifest.json"
```
Comment (degemer) [inline resources/datadog-integrations/project.rb.erb]: Hum, I don't think it's going to work on Windows

File: resources/datadog-integrations/project.rb.erb [hunk 2]
```diff
@@ -1,11 +1,13 @@
+require "./lib/ostools.rb"
+
 name 'dd-check-<%= name %>'
 maintainer 'Datadog Packages <package@datadoghq.com>'
 homepage 'http://www.datadoghq.com'
-install_dir '/opt/datadog-agent'
+install_dir '/opt/datadog-agent/integrations/<%= name %>'
```
Comment (degemer) [inline resources/datadog-integrations/project.rb.erb]: Why this change?

File: resources/datadog-integrations/project.rb.erb [hunk 3]
```diff
@@ -48,6 +50,15 @@ runtime_dependency 'datadog-agent'
...
+if linux?
+  conf_directory = "/etc/dd-agent/conf.d"
+elsif osx?
+  conf_directory = "/opt/datadog-agent/etc"
+elsif windows?
+  conf_directory = "../../extra_package_files/EXAMPLECONFSLOCATION"
+end
+
+extra_package_file "#{install_dir}/../../embedded/lib/python2.7/site-packages/"
+extra_package_file "#{conf_directory}/examples/"
```
Comment (degemer) [inline resources/datadog-integrations/project.rb.erb]: `extra_package_file` doesn't work on Windows (I don't think it fails the build though), see my previous review.

File: resources/datadog-integrations/software.rb.erb [hunk 4]
```diff
@@ -19,37 +21,34 @@ build do
...
+  copy "#{integration_source_folder}/conf.yaml.example", "#{conf_directory}/examples/<%= name %>.yaml.example"
+  copy "#{integration_source_folder}/conf.yaml.default", "#{conf_directory}/examples/<%= name %>.yaml.default"
 
-  # If it is a default, we need to move it as a default
-  if File.exists? "#{project_dir}/<%= name %>/conf.yaml.default"
-    copy "#{integration_source_folder}/conf.yaml.default", "/etc/dd-agent/conf.d/examples/<%= name %>.yaml.default"
-  end
-
-  command "echo \"#{integration_agent_folder}/lib\" > #{install_dir}/embedded/lib/python2.7/site-packages/#{name}.pth"
+  command "echo \"#{integration_agent_folder}/lib\" > #{agent_install_dir}/embedded/lib/python2.7/site-packages/#{name}.pth"
```
Comment (degemer) [inline resources/datadog-integrations/software.rb.erb]: Why?

File: resources/datadog-integrations/software.rb.erb [hunk 5]
```diff
@@ -1,11 +1,13 @@
+require "./lib/ostools.rb"
+require 'json'
+
 name "dd-check-<%= name %>-software"
 default_version "0.0.1"
 
-require 'json'
-dependency 'pip'
 require_relative "<%= PROJECT_DIR %>/resources/datadog-integrations/validate_manifest"
 integration_source_folder = '/<%= integrations_repo %>/<%= name %>'
```
Comment (degemer) [inline resources/datadog-integrations/software.rb.erb]: It won't work on Windows, but maybe Derek is already working on that?

File: Rakefile.rb [hunk 6]
```diff
@@ -17,10 +17,8 @@
 
     sh "rm -rf /#{ENV['INTEGRATIONS_REPO']}"
     sh "git clone https://github.com/DataDog/#{ENV['INTEGRATIONS_REPO']}.git /#{ENV['INTEGRATIONS_REPO']} || true"
-    sh "cd /integrations-core && ls -a && git status"
     sh "cd /#{ENV['INTEGRATIONS_REPO']} && git fetch --all"
-    sh "cd /#{ENV['INTEGRATIONS_REPO']} && git checkout dd-check-#{ENV['INTEGRATION']}-#{integration_branch} ||
-        git checkout #{integration_branch}"
+    sh "cd /#{ENV['INTEGRATIONS_REPO']} && git checkout master"
```
Comment (gmmeyer) [inline Rakefile.rb]: That's reasonable, I'll put it back in

File: resources/datadog-integrations/project.rb.erb [hunk 7]
```diff
@@ -48,6 +50,14 @@ runtime_dependency 'datadog-agent'
...
-extra_package_file '/etc/dd-agent/conf.d/examples/<%= name %>.yaml.example'
+if linux?
+  conf_directory = "/etc/dd-agent/conf.d"
+elsif osx?
+  conf_directory = "/opt/datadog-agent/etc"
+elsif windows?
+  conf_directory = "../../extra_package_files/EXAMPLECONFSLOCATION"
+end
+
+extra_package_file "#{conf_directory}/examples/"
```
Comment (gmmeyer) [inline resources/datadog-integrations/project.rb.erb]: Okay, I'll put it behind an if statement

File: resources/datadog-integrations/software.rb.erb [hunk 8]
```diff
@@ -1,7 +1,9 @@
+require "./lib/ostools.rb"
+require 'json'
+
 name "dd-check-<%= name %>-software"
 default_version "0.0.1"
 
-require 'json'
 dependency 'pip'
 require_relative "<%= PROJECT_DIR %>/resources/datadog-integrations/validate_manifest"
 integration_source_folder = '/<%= integrations_repo %>/<%= name %>'
```
Comment (gmmeyer) [inline resources/datadog-integrations/software.rb.erb]: Yeah Derek probably fixed it in his branch

File: resources/datadog-integrations/software.rb.erb [hunk 9]
```diff
@@ -27,14 +29,8 @@ build do
...
   # Move config files to examples folder.
-  if File.exists? "#{project_dir}/<%= name %>/conf.yaml.example"
-    copy "#{integration_source_folder}/conf.yaml.example", "/etc/dd-agent/conf.d/examples/<%= name %>.yaml.example"
-  end
-
-  # If it is a default, we need to move it as a default
-  if File.exists? "#{project_dir}/<%= name %>/conf.yaml.default"
-    copy "#{integration_source_folder}/conf.yaml.default", "/etc/dd-agent/conf.d/examples/<%= name %>.yaml.default"
-  end
+  copy "#{integration_source_folder}/conf.yaml.example", "#{conf_directory}/examples/<%= name %>.yaml.example"
```
Comment (gmmeyer) [inline resources/datadog-integrations/software.rb.erb]: It does not

**Comment by gmmeyer** [comment]
Yes I'll squash them and I'll address these comments

File: resources/datadog-integrations/software.rb.erb (hunk 4 above)
Comment (degemer) [inline resources/datadog-integrations/software.rb.erb]: Please disregard, I thought you added this line, sorry.

File: resources/datadog-integrations/project.rb.erb (hunk 1 above)
Comment (gmmeyer) [inline resources/datadog-integrations/project.rb.erb]: No, probably not. I'm not sure it does anything

File: resources/datadog-integrations/project.rb.erb (hunk 2 above)
Comment (gmmeyer) [inline resources/datadog-integrations/project.rb.erb]: Because conflicts happen if the agent is the installdir

File: resources/datadog-integrations/project.rb.erb (hunk 3 above)
Comment (gmmeyer) [inline resources/datadog-integrations/project.rb.erb]: Okay I'll put it behind an if statement

File: resources/datadog-integrations/software.rb.erb [hunk 10]
```diff
@@ -44,12 +39,15 @@ build do
...
+
+  if File.exists? "#{integration_source_folder}/requirements.txt"
+    # install requirements
+    install_cmd = "install --target=#{integration_agent_folder}/lib "
+    # Only apply the constraints file if one exists
+    if File.exists? "#{install_dir}/agent/requirements.txt"
+      install_cmd += " -c #{install_dir}/agent/requirements.txt "
+    end
+    install_cmd += " -r #{integration_source_folder}/requirements.txt"
+    pip install_cmd, :base_dir => "#{install_dir}"
```
Comment (degemer) [inline resources/datadog-integrations/software.rb.erb]: base_dir is not needed anymore I think

**Comment by gmmeyer** [comment]
good spot, that's gone now!
